netcontrold/tests/lib/test_util.py

```diff
@@ -262,3 +262,35 @@ def test_rr_cpu_in_numa_positive(self, mock_open):
         out = util.rr_cpu_in_numa()
         expected = [0, 2, 1, 3]
         self.assertEqual(out, expected)
+
+
+class TestUtil_test_thread_creation(TestCase):
+
+    def setUp(self):
+        util.Memoize.forgot = True
+        thread = util.Thread()
```

Comment:
You need to inherit util.Thread like how CtlDThread does in main app. For eg:

class DummyNcdThread(util.Thread):
    def __init__(self, eobj):
        util.Thread.__init__(self, eobj)

    def run(self):
        <set a dummy variable to 1>
       while (not self.ncd_shutdown.is_set()):
          <do some no_op with some sleep interval not to hog cpu>
       return

class TestUtil_test_thread_creation(TestCase):

  def setUp(self):
        util.Memoize.forgot = True

  def check_thread_exist(self):
        stop_event = threading.Event()
        tobj = DummyNcdThread(stop_event)
        tobj.start()
        tobj.stop_event.set()
        tobj.join()
       <assert dummy variable set by thread equal to 1>

  def check_thread_no_exist(self):
     < think of thread failure case >